### PR TITLE
RS:245: Split undo modal from pickers component

### PR DESCRIPTION
**Pull request by blazeque**

### CODE REVIEW

File: vue/components/undo/undo.vue [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    {{ "ripe_commons.pickers.backbutton" | locale }}
+                </a>
+                <span>
+                    {{ "ripe_commons.pickers.limited" | locale }}
+                    {{ "ripe_commons.pickers.back" | locale }}
+                </span>
+            </div>
+        </div>
+    </div>
+</template>
```
Comment (gcandal) [inline vue/components/undo/undo.vue]: add a newline here

File: vue/components/undo/undo.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    cursor: pointer;
+    float: right;
+    font-size: 12px;
+    font-weight: bold;
+    margin-left: 20px;
+    margin-top: -10px;
+    padding: 10px 10px 10px 10px;
+    text-decoration: underline;
+}
+</style>
```
Comment (gcandal) [inline vue/components/undo/undo.vue]: newline

File: bundles/config.en_us.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
     "ripe_commons.modal.select": "Select",
 
     "ripe_commons.pickers.undo": "Undo",
+    "ripe_commons.pickers.backbutton": "Back",
```
Comment (gcandal) [inline bundles/config.en_us.json]: backbutton -> `back_button`  (we separate words with underscore)

**Comment by gcandal** [comment]
Looks good. Ask @Benedito0 to help you set up RIPE Retail so that you can test this change there.

When that's done, please most a gif of White + Retail working with this change.

File: vue/components/undo/undo.vue (hunk 1 above)
Comment (blazeque) [reply vue/components/undo/undo.vue]: New code version commited

**Comment by gcandal** [comment]
Don't worry about "warning" about changes in the comments, assigning me the
PR is enough for me to know you made the requested changes and it's ready
to look at again :)

Thanks!

A terça, 5 de nov de 2019, 12:27, Daniel Blazeque <notifications@github.com>
escreveu:

**Comment by gcandal** [comment]
The code is good to go, we're just missing the GIFs (Retail + White) to confirm that nothing got broken.

**Comment by blazeque** [comment]
**Ripe-Retail**

Screen

**Ripe-White**

Screen

File: vue/components/pickers/pickers.vue [hunk 4]
```diff
@@ -308,39 +298,6 @@
     border-width: 4px;
 }
 
-.pickers .message-undo-container {
```
Comment (gcandal) [inline vue/components/pickers/pickers.vue]: Where did this CSS go?

**Comment by gcandal** [comment]
@blazeque I made a few small changes, please check that everything is still working.

**Comment by joamag** [comment]
@gcandal fix the conflicts

**Comment by blazeque** [comment]
Test Ripe-Retail after CSS's changes

ezgif-4-c2aec3d1638a

**Comment by joamag** [comment]
@gcandal let's call this `restrictions-alert` just as defined in https://github.com/ripe-tech/ripe-white/issues/90

File: bundles/config.en_us.json [hunk 5]
```diff
@@ -6,9 +6,10 @@
     "ripe_commons.modal.apply": "Apply",
     "ripe_commons.modal.select": "Select",
 
-    "ripe_commons.pickers.undo": "Undo",
-    "ripe_commons.pickers.limited": "This option is limited to specific material or colour combinations.",
-    "ripe_commons.pickers.back": "Do you want to go back?",
+    "ripe_commons.restrictionsAlert.restrictionsAlert": "Undo",
```
Comment (gcandal) [inline bundles/config.en_us.json]: Should be `restrictions_alert` (look a few lines below and you'll find `add_initials`)

File: js/base/main.js [hunk 6]
```diff
@@ -263,8 +263,7 @@ class RipeCommonsMainPlugin extends RipeCommonsPlugin {
                 this.$bus.bind("initials_change", initialsExtra => {
                     self.ripe.setInitialsExtra(initialsExtra);
                 });
-                this.$bus.bind("undo", () => self.ripe.undo());
```
Comment (blazeque) [inline js/base/main.js]: revert

File: bundles/config.en_us.json [hunk 7]
```diff
@@ -6,9 +6,10 @@
     "ripe_commons.modal.apply": "Apply",
     "ripe_commons.modal.select": "Select",
 
-    "ripe_commons.pickers.undo": "Undo",
-    "ripe_commons.pickers.limited": "This option is limited to specific material or colour combinations.",
-    "ripe_commons.pickers.back": "Do you want to go back?",
+    "ripe_commons.restrictionAlerts.restrictionAlerts": "Undo",
```
Comment (blazeque) [inline bundles/config.en_us.json]: restrictions_alert

File: vue/components/index.js [hunk 8]
```diff
@@ -14,5 +14,6 @@ export { personalization } from "./personalization.vue";
 export { size } from "./size.vue";
 export { thumbnail } from "./thumbnail.vue";
 export { thumbnails } from "./thumbnails.vue";
+export { restrictionAlerts } from "./restrictionAlerts";
```
Comment (blazeque) [inline vue/components/index.js]: restrictionsAlert

File: vue/components/pickers/pickers.vue [hunk 9]
```diff
@@ -394,6 +345,16 @@ body.mobile .button-scroll-colors {
     height: 100px;
     width: 50px;
 }
+
+.restrictionAlerts .message-restrictionAlerts-container .message-restrictionAlerts.visible {
```
Comment (blazeque) [inline vue/components/pickers/pickers.vue]: restrictions-alert

File: vue/components/restrictionAlerts/restrictionAlerts.vue [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    },
+    mounted: function() {
+        this.$bus.bind("restrictions", (changes, newPart) => {
+            this.visible = changes.length > 0;
+        });
+    },
+    methods: {
+        restrictionAlerts() {
+            this.visible = false;
+            this.$bus.trigger("restrictionAlerts");
```
Comment (blazeque) [inline vue/components/restrictionAlerts/restrictionAlerts.vue]: undo

**Comment by gcandal** [comment]
@blazeque please merge with master

**Pull request merged**